packages/apollo-federation/src/service/extractFederationResolvers.ts

```diff
@@ -0,0 +1,47 @@
+import { GraphQLSchema, isObjectType } from 'graphql';
+import { GraphQLResolverMap } from 'apollo-graphql';
+import {
+  GraphQLReferenceResolver,
+  ResolvableGraphQLObjectType,
+} from '../types';
+
+function extractFederationResolversForType(
+  type: ResolvableGraphQLObjectType,
+): { __resolveReference: GraphQLReferenceResolver } | undefined {
+  if (type.resolveReference) {
+    return { __resolveReference: type.resolveReference };
+  }
+
+  if (isObjectType(type)) {
+    const fields = type.getFields();
+
+    if (fields.__resolveReference) {
```

Comment:
I had it as a field here mainly because my implementation on `type-graphql` would create a `@FieldResolver()` `__resolveReference` method which gets added as a field then removed here.  But I made my issue in that repository for my PR regarding the fact that it doesn't make sense as a field resolver because the signature is different anyway, so I agree.  It's just that most libraries could be federated easily by adding it as a field, but `all` libraries should adhere to extensions when it lands.  This entire PR and the federation resolver code can be cleaned up a ton when `extensions` lands.  I can't wait :).  I can remove this though.  You can still make a schema federated by:

```typescript
schema.getType('User').resolveReference = /* ... */;
```